From 85783e8cbba03804e13b3d314112df6b089ed2c0 Mon Sep 17 00:00:00 2001
From: Eric Astor <epastor@google.com>
Date: Fri, 22 Nov 2019 11:07:15 -0500
Subject: [PATCH] Fix exports for 32-bit GNU assembler files targeting Windows.

The incorrect exports can cause linker issues (at least when using LLVM's lld-link) claiming that these primitives are not available.
---
 src/asm/jump_i386_ms_pe_gas.asm  | 2 +-
 src/asm/make_i386_ms_pe_gas.asm  | 2 +-
 src/asm/ontop_i386_ms_pe_gas.asm | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/asm/jump_i386_ms_pe_gas.asm b/src/asm/jump_i386_ms_pe_gas.asm
index 8512a3d0..4c623006 100644
--- a/src/asm/jump_i386_ms_pe_gas.asm
+++ b/src/asm/jump_i386_ms_pe_gas.asm
@@ -114,4 +114,4 @@ _jump_fcontext:
     jmp *%ecx
 
 .section .drectve
-.ascii " -export:\"jump_fcontext\""
+.ascii " -export:\"_jump_fcontext\""
diff --git a/src/asm/make_i386_ms_pe_gas.asm b/src/asm/make_i386_ms_pe_gas.asm
index dcb77000..f2e3b689 100644
--- a/src/asm/make_i386_ms_pe_gas.asm
+++ b/src/asm/make_i386_ms_pe_gas.asm
@@ -144,4 +144,4 @@ finish:
 .def	__exit;	.scl	2;	.type	32;	.endef  /* standard C library function */
 
 .section .drectve
-.ascii " -export:\"make_fcontext\""
+.ascii " -export:\"_make_fcontext\""
diff --git a/src/asm/ontop_i386_ms_pe_gas.asm b/src/asm/ontop_i386_ms_pe_gas.asm
index 41f15f5b..8bfe93c6 100644
--- a/src/asm/ontop_i386_ms_pe_gas.asm
+++ b/src/asm/ontop_i386_ms_pe_gas.asm
@@ -122,4 +122,4 @@ _ontop_fcontext:
     jmp  *%ecx
 
 .section .drectve
-.ascii " -export:\"ontop_fcontext\""
+.ascii " -export:\"_ontop_fcontext\""
